### PR TITLE
created new auth context, wrapped layout in auth context, export hook…

### PR DESCRIPTION
**Pull request by dionyichia**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

created new auth context, wrapped layout in auth context, exported hook and session and used it in auth page. verified with new signup and sign in.

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: utils/AuthProvider.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  };
+
+  const signInWithEmail = async (email: string, password: string) =>
+    supabase.auth.signInWithPassword({
+      email,
+      password,
+    });
+
+  const signOut = () => {
+    supabase.auth.signOut();
```
Comment (me-liu) [inline utils/AuthProvider.ts]: make sure to use await in this functions

File: utils/AuthProvider.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      signOut,
+    }),
+    [session],
+  );
+
+  return React.createElement(
+    AuthContext.Provider,
+    { value: authContextValue },
+    children,
+  );
```
Comment (kylezryr) [inline utils/AuthProvider.ts]: you can make this a little cleaner by returning `<AuthContext.Provider value={authContextValue}>{children}</AuthContext.Provider>` to avoid using `React.createElement`, just looks cleaner imo but same functionality

**Review by kylezryr — CHANGES_REQUESTED** [review]
looks good dion! just a small change to make along with melissa's comments but great work

File: utils/AuthProvider.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    });
+  }, []);
+
+  const signUp = async (email: string, password: string) =>
+    supabase.auth.signUp({
+      email,
+      password,
+    });
+
+  const signIn = (newSession: Session | null) => {
```
[the remark this answers is not in the document]
Comment (dionyichia) [reply utils/AuthProvider.ts]: alrighty done

File: utils/AuthProvider.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const [session, setSession] = useState<Session | null>(null);
+
+  useEffect(() => {
+    supabase.auth.getSession().then(({ data: { session: newSession } }) => {
+      setSession(newSession);
+    });
+
+    supabase.auth.onAuthStateChange((_event, newSession) => {
+      setSession(newSession);
+    });
```
[the remark this answers is not in the document]
Comment (dionyichia) [reply utils/AuthProvider.ts]: alrighty
